transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpireDateUtils.java

```diff
@@ -137,16 +141,20 @@ boolean isValidMonth(@Nullable String monthString) {
     }
 
     // create Date from input to compare with current Date
-    private Calendar getInputDate(@NonNull final String input) {
+    private Calendar getInputDate(@NonNull final String input) throws IllegalArgumentException {
         Calendar inputCalendar = Calendar.getInstance();
         inputCalendar.set(Calendar.MONTH, getInputMonth(input));
         inputCalendar.set(Calendar.YEAR, getInputYear(input));
         return inputCalendar;
     }
 
     // get month from input Date
-    private Integer getInputMonth(String input) {
-        return Integer.valueOf(input.substring(6, 7)) - 1;
+    private int getInputMonth(@NonNull final String input) throws IllegalArgumentException {
+        String monthString = input.substring(5);
+        if (isValidMonth(monthString)) {
```

Comment:
May i know why we are trusting string operations instead of predefined Date parsing classes(like Calender, Date, SimpleDateFormat etc). In my point of view more code, attention needed for date parsing using string operations.

Reply:
Good question. We should have done it using parsers like you mentioned, I've updated the code with the suggested approach. Note that there are plenty of room for improvements in that class but I won't refarctor it in this PR